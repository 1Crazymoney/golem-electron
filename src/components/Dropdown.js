import React from 'react';

export default class Dropdown extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            listVisible: false,
            selectedIndex: props.selected
        }
    }

    componentDidMount() {
        const {handleChange, list, selected} = this.props
        if (handleChange && list[selected]) {
            handleChange(list[selected].name, true)
        }
    }

    componentWillUpdate(nextProps, nextState) {
        if (nextProps.selected !== this.props.selected) {
            console.log("NEXT_PROPS", nextProps.selected)
            this.setState({
                selectedIndex: nextProps.selected
            });
        }
    }

    select(item, index) {
        this.setState({
            listVisible: false,
            selectedIndex: index
        });
        this.props.handleChange(item.name)
    }

    show() {
        this.setState({
            listVisible: !this.state.listVisible
        });
    }

    hide() {
        this.setState({
            listVisible: false
        });
    }

    _handleManageModal(list) {
        this.props.manageHandler()
        this.hide()
    }

    componentWillUnmount() {}

    renderListItems(list) {
        const {selectedIndex} = this.state
        return list.map((item, index) => <div key={index.toString()} className="item__dropdown" onClick={this.select.bind(this, item, index)}>
        <span className={selectedIndex === index ? 'selected' : ''}>{item.name}</span>
      </div>)
    }

    render() {
        const {selected, list, presetManager, disabled} = this.props
        const {listVisible, selectedIndex} = this.state
        return (
            <div className="dropdown-container">
                <div className="dropdown-display" onClick={!disabled && ::this.show}>
                    <span>{list[selectedIndex] ? list[selectedIndex].name : 'Not loaded'}</span>
                </div>
                {listVisible && <div className="dropdown-list">
                                    <div className="item-container__dropdown">
                                        {list.length > 0 ? this.renderListItems(list) : <span className="info__dropdown">There's no preset. You must create first.</span>}
                                    </div>
                                    {list.length > 0 && presetManager && <div className="option__dropdown">
                                        <div className="item__dropdown" onClick={::this._handleManageModal}>
                                            <span>Manage presets</span>
                                        </div>
                                    </div>}
                                </div>}
            </div>
        );
    }
}
