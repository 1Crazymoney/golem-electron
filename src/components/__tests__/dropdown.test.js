jest.unmock('../Dropdown')

import React from 'react'
import ReactDOM from 'react-dom'
import TestUtils from 'react-dom/test-utils'
import sinon from 'sinon'
import Dropdown from '../Dropdown'


describe('<Dropdown/>', () => {

    const mockList = ['Golem', 'Network']

    it('should render a dropdown component', () => {
        const wrapper = shallow(
            <Dropdown 
                list={mockList} 
                selected={0} 
                handleChange={jest.fn} />
        )
        expect(wrapper).toMatchSnapshot()
    })

});