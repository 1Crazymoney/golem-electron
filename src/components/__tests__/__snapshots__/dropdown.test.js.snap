// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> should render a dropdown component 1`] = `
<div
  className="dropdown-container"
>
  <div
    className="dropdown-display"
    onClick={[Function]}
  >
    <span />
  </div>
</div>
`;
